assets/js/base/hocs/with-reviews.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * External dependencies
+ */
+import { Component } from 'react';
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { speak } from '@wordpress/a11y';
+
+/**
+ * Internal dependencies
+ */
+import { getReviews, getOrderArgs } from '../../blocks/reviews/utils';
+
+const withReviews = ( OriginalComponent ) => {
+	return class WrappedComponent extends Component {
+		constructor() {
+			super( ...arguments );
+
+			this.state = {
+				error: null,
+				loading: false,
+				overwrittenArgs: {},
+				reviews: [],
+				totalReviews: 0,
+			};
+
+			this.getReviews = this.getReviews.bind( this );
+			this.appendReviews = this.appendReviews.bind( this );
+			this.onChangeArgs = this.onChangeArgs.bind( this );
+			this.updateListOfReviews = this.updateListOfReviews.bind( this );
+			this.setError = this.setError.bind( this );
+		}
+
+		componentDidMount() {
+			this.getReviews();
+		}
+
+		getDefaultArgs() {
+			const { attributes } = this.props;
+			const { overwrittenArgs } = this.state;
+			const { order, orderby } = getOrderArgs( attributes.orderby );
+			const { productId, reviewsOnPageLoad } = attributes;
+
+			return {
+				order,
+				orderby,
+				per_page: reviewsOnPageLoad,
+				product_id: productId,
+				...overwrittenArgs,
+			};
+		}
+
+		getReviews() {
```

Comment:
It seems the only reason this is present is to pass it along as a way for wrapped components to trigger getting reviews.  Instead, I think you should update a `reviews` property in the state from any updates and only pass along the retrieved review data.  Passing along this function doesn't really decouple the data retrieval that much.  Ideally, the you want `withReviews` to handle all the retrieval and updating of reviews and just pass along the retrieved reviews to wrapped presentational components.  

For the `debounced` behaviour you have for the editor block implementation, you can control that via a `debounced` boolean prop fed into the hoc.  And that can control whether it does debounced getReviews on the `componentDidUpdate` lifecycle event (which should be in the hoc, not the presentational components).

Reply:
Right, the reason this is passed to the component was to simplify the frontend block so it didn't need the `debounce` dependency and didn't make any unncessary computations on `componentDidUpdate`.

`componentDidUpdate()` is just a couple of checks, so I don't think there is much performance difference, I moved it to the HOC as suggested. :+1: 

Regarding the debounce dependency, I moved it to `edit.js` and passed it as a prop. I would like to avoid loading dependencies which are not used in the frontend.

Let me know how it looks: 5d7bac1001ae56a3a0f73e32672a9c068a6115ae.